cmd/main.go

```diff
@@ -139,3 +161,101 @@ func main() {
 
 	os.Exit(0)
 }
+
+func runMultishareReconciler(driverConfig *driver.GCFSDriverConfig) {
+	config, err := buildConfig(*kubeconfig)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	fsClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	factory := fsInformers.NewSharedInformerFactory(fsClient, resyncPeriod)
+	coreFactory := informers.NewSharedInformerFactory(kubeClient, resyncPeriod)
+	sharescheme.AddToScheme(scheme.Scheme)
+
+	recon := reconciler.NewMultishareReconciler(
+		fsClient,
+		driverConfig,
+		factory.Multishare().V1alpha1().ShareInfos(),
+		factory.Multishare().V1alpha1().InstanceInfos(),
+		coreFactory.Storage().V1().StorageClasses().Lister(),
+	)
+
+	if err := ensureCustomResourceDefinitionsExist(fsClient); err != nil {
```

Comment:
if we move the runMultishareReconciler() to GCFSDriver.run(), the ensure CRD  exists can be moved out and remain in cmd/main.go under the `*runController && *enableMultishare && *enableStateful` check. Thus by the time we reach the reconciler start logic, we are ensured the CRD definition exits.

Thoughts on this restructure?

Reply:
the `ensureCustomResourceDefinitionsExist` function needs to have a client passed in, that means the client initiation code needs to also live in main. then main.go it might be a little messy. What do you think?

Reply:
Ack let's keep it as is. FYI the kubeclient would be needed for 100shares to create PV informer cache. so let's keep that in main